web/src/compose_recipient.ts

```diff
@@ -403,17 +405,29 @@ export function update_topic_displayed_text(topic_name = "", has_topic_focus = f
     $topic_not_mandatory_placeholder.hide();
     $("#compose_recipient_box").removeClass("disabled");
 
-    if (!stream_data.can_use_empty_topic(compose_state.stream_id())) {
+    if (!stream_data.can_use_empty_topic(stream_id)) {
         $input.attr("placeholder", $t({defaultMessage: "Topic"}));
         // When topics are mandatory, no additional adjustments are needed.
         // Also, if the recipient in the compose box is not selected, the
         // placeholder will always be "Topic" and never "general chat".
         return;
     }
 
+    const cannot_create_new_topics_in_stream =
```

Comment:
I think, after the `if (!stream_data.can_use_empty_topic(stream_id)` block, `stream_id` can't be `undefined` ?

We can add an assertion and then code will be much simplified